File: Data Structure/Array or Matrix/Medium/LET_no849_Maximize Distance to Closest Person.py
class Solution:
    def maxDistToClosest(self, seats: List[int]) -> int:
        res, last = 0, -1
        
        for i in range(len(seats)):
            if seats[i]:
                res = max(res, i if last < 0 else (i - last) // 2)
                last = i
        
        return max(res, len(seats) - 1 - last)

File: Data Structure/Array or Matrix/Medium/LET_no849_Maximize Distance to Closest Person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int res = 0, pre = -1, n = seats.size();
        for (int i = 0; i < n; i++) {
            if (seats[i]) {
                res = max(res, pre < 0 ? i : (i - pre) / 2);
                pre = i;
            }
        }
        return max(res, n - 1 - pre);
    }
};